File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        //eulb si yks eth
        reverse(s.begin(),s.end()); 
        string answer = ""; 
        int start = 0;  
        while(start < s.length()){
            //cout << start << ' '; 
            while(start < s.length() && isspace(s[start])) start++; 
            int end = start; 
            while(end < s.length() && (isalpha(s[end]) || isdigit(s[end]))) end++; 
            
            string tmp = s.substr(start, end - start); 
            reverse(tmp.begin(),tmp.end());
            if(tmp.length() > 0) answer += tmp + ' '; 
            start = end; 
        }
        answer.pop_back(); 
        return answer; 
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s[::-1] 
        start = 0
        end = 0 
        answer = "" 
        while(end < len(s)):
            while(start < len(s) and s[start].isspace()) : start += 1
            end = start 
            while(end < len(s) and s[end].isalnum()) : end += 1 
            
            tmp = s[start:end]
            tmp = tmp[::-1]
            if len(tmp) > 0 : answer += tmp + " "
            start = end 
        
        answer = answer[:len(answer)-1]
        return answer